django/contrib/admin/static/admin/js/cancel.js

```diff
@@ -1,13 +1,27 @@
-(function($) {
+(function() {
     'use strict';
-    $(function() {
-        $('.cancel-link').on('click', function(e) {
-            e.preventDefault();
-            if (window.location.search.indexOf('&_popup=1') === -1) {
-                window.history.back(); // Go back if not a popup.
-            } else {
-                window.close(); // Otherwise, close the popup.
-            }
+
+    // Call function fn when the DOM is loaded and ready. If it is already
+    // loaded, call the function now.
+    // http://youmightnotneedjquery.com/#ready
+    function ready(fn) {
+        if (document.readyState !== 'loading') {
+            fn();
+        } else {
+            document.addEventListener('DOMContentLoaded', fn);
+        }
+    }
+
+    ready(function() {
+        document.querySelectorAll('.cancel-link').forEach(function(el) {
+            el.addEventListener('click', function(e) {
+                e.preventDefault();
+                if (window.location.search.indexOf('&_popup=1') === -1) {
+                    window.history.back(); // Go back if not a popup.
+                } else {
+                    window.close(); // Otherwise, close the popup.
+                }
+            });
         });
     });
```

Comment:
The callback could be shared across the `.cancel-link` elements, instead of defining a new anonymous function for each.

```suggestion
    ready(function() {
        function handleClick(event) {
            e.preventDefault();
            if (window.location.search.indexOf('&_popup=1') === -1) {
                window.history.back(); // Go back if not a popup.
            } else {
                window.close(); // Otherwise, close the popup.
            }
        }
        document.querySelectorAll('.cancel-link').forEach(function(el) {
            el.addEventListener('click', handleClick);
        });
    });
```